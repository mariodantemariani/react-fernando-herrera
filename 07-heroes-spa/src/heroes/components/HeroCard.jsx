import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import "animate.css";

const CharactersByHero = ({ characters, alter_ego }) => {
  return characters === alter_ego ? (
    <></>
  ) : (
    <p className="card-text">{characters}</p>
  );
};

export const HeroCard = ({
  id,
  superhero,
  alter_ego,
  first_appearance,
  characters,
}) => {
  const heroImageUrl = `/assets/heroes/${id}.jpg`;
  const heroUrl = `/hero/${id}`;

  ///const charactersByHero = <p className="card-text">{characters}</p>;

  return (
    <div className="col animate__animated animate__fadeIn">
      <div className="card ms-3" style={{ maxWidth: 540 }}>
        <div className="row no-gutters">
          <div className="col-md-4">
            <img src={heroImageUrl} className="card-img" alt={superhero} />
          </div>
          <div className="col-md-8">
            <div className="card-body">
              <h5 className="card-title">{superhero}</h5>
              <p className="card-text">{alter_ego}</p>
              {/* {alter_ego !== characters && charactersByHero} */}
              <CharactersByHero characters={characters} alter_ego={alter_ego} />
              <p className="card-text">
                <small className="text-muted">{first_appearance}</small>
              </p>
              <Link to={heroUrl}>Más...</Link>
              {/* <button className="btn btn-outline-primary mt-5" >Ver más...</button> */}
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

CharactersByHero.propTypes = {
  characters: PropTypes.string.isRequired,
  alter_ego: PropTypes.string.isRequired,
};

HeroCard.propTypes = {
  id: PropTypes.string.isRequired,
  superhero: PropTypes.string.isRequired,
  alter_ego: PropTypes.string.isRequired,
  first_appearance: PropTypes.string.isRequired,
  characters: PropTypes.string.isRequired,
  publisher: PropTypes.string.isRequired,
};
