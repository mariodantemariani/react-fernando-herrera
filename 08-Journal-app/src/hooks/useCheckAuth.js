import { useDispatch, useSelector } from "react-redux";

import { useEffect } from "react";
import { onAuthStateChanged } from "firebase/auth";
import { FirebaseAuth } from "../firebase/config";
import { login, logout } from "../store/auth";
import { startLoadingNotes } from "../store/journal";

export const useCheckAuth = () => {
  console.log("useCheckAuth");
  const { status } = useSelector((state) => state.auth);
  const dispatch = useDispatch();

  //disparar un efecto para disparar si la persona esta autenticada o no.
  //si esta autenticada redirigir a al main o si no redirigir a Auth
  //usando firebase
  useEffect(() => {
    onAuthStateChanged(FirebaseAuth, async (user) => {
      if (!user) return dispatch(logout());

      const { uid, displayName, email, photoURL } = user;

      dispatch(login({ uid, displayName, email, photoURL }));
      dispatch(startLoadingNotes());
    });
  }, [status]);

  return status;
};
