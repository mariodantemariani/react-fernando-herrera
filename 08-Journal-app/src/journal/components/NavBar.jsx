import { AppBar, Grid, IconButton, Toolbar, Typography } from "@mui/material";
import { LogoutOutlined, MenuOutlined } from "@mui/icons-material";

export const NavBar = ({ drawerWidth }) => {
  return (
    <AppBar
      position="fixed"
      sx={{
        width: { sm: `calc(100% - ${drawerWidth}px)` },
        ml: { sm: `${drawerWidth}px` },
      }}
    >
      <Toolbar>
        <IconButton
          color="inherit"
          aria-label="menu"
          edge="start"
          sx={{ mr: 2, display: { sm: "none" } }}
        >
          <MenuOutlined />
        </IconButton>
        <Grid
          container
          direction="row"
          justifyContent="space-between"
          alignItems="center"
        >
          <Typography variant="h6" noWrap component="div" sx={{ flexGrow: 1 }}>
            JournalApp
          </Typography>
          <IconButton color="error">
            <LogoutOutlined />
          </IconButton>
        </Grid>
      </Toolbar>
    </AppBar>
  );
};
