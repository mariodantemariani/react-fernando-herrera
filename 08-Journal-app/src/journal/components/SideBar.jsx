import { useDispatch, useSelector } from "react-redux";

import {
  Box,
  Divider,
  Drawer,
  Grid,
  List,
  ListItem,
  ListItemButton,
  ListItemIcon,
  ListItemText,
  Toolbar,
  Typography,
} from "@mui/material";

import { SideBarItem } from "./SideBarItem";

export const SideBar = ({ drawerWidth = 240 }) => {
  const { displayName } = useSelector((state) => state.auth);
  const { notes } = useSelector((state) => state.journal);

  return (
    <Box
      component={"nav"}
      sx={{ width: { sm: drawerWidth }, flexShrink: { sm: 0 } }}
    >
      <Drawer
        variant="permanent"
        open
        sx={{
          display: { xs: "block" },
          "& .MuiDrawer-paper": { boxSizing: "border-box", width: drawerWidth },
        }}
      >
        <Toolbar>
          <Typography variant="h6" noWrap component="div">
            {displayName}
          </Typography>
        </Toolbar>
        <Divider />

        <List>
          {notes.map((note) => (
            <SideBarItem key={note.id} {...note} />
            // <ListItem key={note.id} disablePadding>
            //   <ListItemButton>
            //     <ListItemIcon>
            //       <TurnedInNot />
            //     </ListItemIcon>
            //     <Grid container>
            //       <ListItemText primary={note.title} />
            //       <ListItemText
            //         secondary={"Exercitation cillum irure elit consectetur."}
            //       />
            //     </Grid>
            //   </ListItemButton>
            // </ListItem>
          ))}
        </List>
      </Drawer>
    </Box>
  );
};
