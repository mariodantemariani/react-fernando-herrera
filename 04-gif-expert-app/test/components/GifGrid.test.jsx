import { render, screen } from "@testing-library/react";
import { GifGrid } from "../../src/components/GifGrid";
import { useFetchGifs } from "../../src/hooks/useFetchGifs";

jest.mock("../../src/hooks/useFetchGifs");

describe("Pruebas en GifGrid", () => {
  const category = "One Punch";

  test("Should match with snapshoot", () => {
    useFetchGifs.mockReturnValue({
      images: [],
      isLoading: true,
    });

    const { container } = render(<GifGrid category={category} />);
    //asserts
    expect(container).toMatchSnapshot();
  });

  test("Should shown the loading", () => {
    useFetchGifs.mockReturnValue({
      images: [],
      isLoading: true,
    });

    render(<GifGrid category={category} />);
    expect(screen.getByText("Cargando...")).toBeTruthy();
    expect(screen.getByText(category)).toBeTruthy();
  });

  test("Should shown the items when images are loaded", () => {
    useFetchGifs.mockReturnValue({
      images: [],
      isLoading: true,
    });

    render(<GifGrid category={category} />);
    expect(screen.getByText(category)).toBeTruthy();
  });

  test("Should shown the items when images are loaded-2", () => {
    const gifs = [
      {
        id: "ABC",
        title: "Saitama",
        url: "https://localhost/saitama.jpg",
      },
      {
        id: "123",
        title: "Goku",
        url: "https://localhost/goku.jpg",
      },
    ];

    useFetchGifs.mockReturnValue({
      images: gifs,
      isLoading: false,
    });

    render(<GifGrid category={category} />);
    expect(screen.getAllByRole("img").length).toBe(2);
  });
});
