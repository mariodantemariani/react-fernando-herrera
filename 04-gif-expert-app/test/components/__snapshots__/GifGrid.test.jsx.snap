// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid Should match with snapshoot 1`] = `
<div>
  <h3>
    One Punch
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
